transport/http/agent.go

```diff
@@ -0,0 +1,123 @@
+package http
+
+import (
+	"net/http"
+	"sync"
+
+	"go.uber.org/yarpc/transport"
+	"go.uber.org/yarpc/transport/internal/errors"
+	"go.uber.org/yarpc/transport/peer/hostport"
+)
+
+// Agent keeps track of http peers and the associated client with which the peer will call into.
+type Agent struct {
+	sync.Mutex
+
+	client    *http.Client
+	peerNodes map[string]*peerNode
+}
+
+// peerNode keeps track of a HostPortPeer and any subscribers referencing it
+type peerNode struct {
+	peer       *hostport.Peer
+	references map[transport.PeerSubscriber]struct{}
+}
+
+// NewDefaultAgent creates an http agent with the default parameters
+func NewDefaultAgent() *Agent {
+	return NewAgent(&defaultConfig)
+}
+
+// NewAgent creates a new http agent for managing peers and sending requests
+func NewAgent(cfg *outboundConfig) *Agent {
```

Comment:
indeed, this was a temporary change to reuse the outboundConfig until I could rename it the agentConfig